### PR TITLE
fix: add missing cython version pin to the build system

### PR DESCRIPTION
**Pull request by bdraco**
related comment: https://github.com/Bluetooth-Devices/dbus-fast/pull/264#issuecomment-2334550288

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -101,5 +101,5 @@ module = "docs.*"
 ignore_errors = true
 
 [build-system]
-requires = ['setuptools>=65.4.1', 'wheel', 'Cython', "poetry-core>=1.0.0"]
+requires = ['setuptools>=65.4.1', 'wheel', 'Cython>=0.29.32,<3.1.0', "poetry-core>=1.0.0"]
```
Comment (jameshilliard) [inline pyproject.toml:104]: The [error](https://github.com/Bluetooth-Devices/dbus-fast/pull/264#issuecomment-2334550288) was hit with Cython version `0.29.37`, so I don't think this is accurate.

Comment (bdraco) [reply pyproject.toml:104]: Yeah it looks like we need 3+

Comment (bdraco) [reply pyproject.toml:104]: https://github.com/Bluetooth-Devices/dbus-fast/pull/311

**Comment by bdraco** [comment]
https://github.com/Bluetooth-Devices/dbus-fast/releases/tag/v2.24.2 should appear in a hour or so

**Pull request merged**